ldms/python/ldmsd/parser_util.py

```diff
@@ -169,27 +169,35 @@ def parse_yaml_bool(bool_):
         return False
 
 def perm_handler(perm_str):
-    if perm_str is None or type(perm_str) is int:
+    if perm_str is None:
         return perm_str
-    nperm = "0"
-    perm_str = perm_str.split('-')
-    if len(perm_str) == 1:
+    if type(perm_str) is not str:
+        raise TypeError(f'Error: YAML "perms" value must be a string')
+    nperm = '0'
+    if len(perm_str.split('-')) > 1:
+        if len(perm_str) != 9:
+            raise ValueError(f'Error: Parsing YAML permisson string "{perm_str}"\n'
+                             f'When using unix-like vernacular to configure permissions you must account for read/write in user/group/global\n'
+                             f'Allowed format: (r|-)(w|-)-(r|-)(w|-)-(r|-)(w|-)-')
+        i = 0
+        x = 0
+        for ch in perm_str:
+            if i == 2:
+                nperm += str(x)
+                x = 0
+                i = 0
+                continue
+            if ch == 'r':
+                x += 4
+            if ch == 'w' or ch == '+':
+                x += 2
+            i += 1
+    else:
         try:
-            z = int(perm_str[0])
-            return perm_str[0]
+            z = int(perm_str)
```

Comment:
I would suggest that we always treat the decimal string as octal:

```
  z = int(perm_str, base=8)
  perm_str = '0'+oct(z)[:2]
```

Then the user doesn't have to be as careful with octal. All of these would work from a string:

```
 perm: "640"
 perm: "0640"
 perm: "0o640"
```

I don't think we need to support non-octal decimals for this.

Reply:
I think this would be staggeringly confusing for the user. No need to be clever. Support something we can document. Let's support what strtoul(x, NULL, 0) would do and then refer to this in our doc.

Reply:
> I think this would be staggeringly confusing for the user. No need to be clever. Support something we can document. Let's support what strtoul(x, NULL, 0) would do and then refer to this in our doc.

I can't see how  interpreting "640" as rw-r-----, what basically every user would expect for permissions bits, is "staggeringly confusing" or even particularly clever.  Could you elaborate on why that would be? Do you know any people that regularly do permission bits in base 10?

I think using base 8 translation for things that are octal like this is pretty ordinary. I don't don't claim to have thought that up.

I think most users would be rather confused to put in "640" and get, at best, "r--------". (Is there even any error checking for numbers with bits out of the expected range?)

Reply:
I'll point out too, that "chmod" doesn't accept base-10 numbers, only base-8. Decimals digits are assumed to be base-8 as I have proposed.

Likewise, when dealing with file permissions in the "find" command, it only interprets digits as base-8.

The more that I look around, the only staggeringly confusing thing seems to be that ldmsd has ever accepted perms that were _not_ in octal form.

Reply:
Yes @morrone, there is a third parameter to strtoul, which is the base. I was talking about the python format "0o123".isdecimal() which is what you proposed and returns false. This is a $1 problem on which OGC has spent $1000. Nick is working on adding the checking for any character other than 'r', 'w', and '-'...which is 1 line of code. I agree with you that specifying permissions in base 10 doesn't make much sense....which leaves...exactly what was suggested.

Reply:
OK, you didn't say that. I'm just fine with "0o123".isdecimal() not working. That wasn't at all the driving issue. The main issue was making it octal-only rather than base 10. I supplied most of the code to make the changes that I suggested, and if it is a monetary issue to integrate them further, you could say that as well, and I'll supply the full patch.

Reply:
or the regex:

```
  (0o)?\d+
```

instead of .decimal()